Audio Parameters Were Not Serializable

Bobtheapple opened the pull request:
PlaySoundMessage attempts to serialize the AudioSourceParameters it receives through the send/sendToAll/sendToNearbyPlayers methods as part of its sending messages to clients.  However, because AudioSourceParameters was using nullable variables and non-serialized get/setters it could not be serialized due to the limitations of the engine.  This resulted in all AudioSourceParameters being ignored and reset to the default null values.  

### Purpose
This fix changes the nullable variables in AudioSourceParameters to normal variables, and instead uses the value of -1 to represent a "default" value.  All code that references AudioSourceParameters using null values has been changed to reflect this.  This change permits all related already existing code to function as intended.  This fix also adjusts ShakeParameters to also be serializable as well, as it was it always produced the same, default shake.

Fixes #5981

### Notes:
Note that this bug impacted almost all code that made changes to pitch, volume, pan, spread, and spatial blending, as any code that played sound through PlaySoundMessage was being reset to default values.  Because of this, many sounds are changed with this fix.

Bobtheapple committed:
Audio Parameters Were Not Serializable

Because of the use of nullable variables and get/setters, AudioSourceParameters and ShakeParameters could not be serialized, making PlaySoundMessage not work as intended.

Aranclanos commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -226,16 +225,12 @@ private void OnSceneChange(Scene oldScene, Scene newScene)
-				SoundSpawns.Add(soundSpawnToken, ToReturn);
-			}
+			ToReturn.Token = soundSpawnToken;
+			SoundSpawns.Add(soundSpawnToken, ToReturn);
```
could you revert this change? it's from today

Bobtheapple committed:
Aranclanos's fixes

Bobtheapple committed:
Changed Audio Parameters from Classes to Structs

This is a large change that significantly alters how AudioSourceParameters, ShakeParameters, and SoundManager's ApplyAudioSourceParameters works.  An additional method ForceAudioSourceParameters was added for better control.

Also, MobExplore was calling SoundManager incorrectly.

Bobtheapple committed:
Muffled Was The Default Audio Mixer

This removed undefined from the audio mixer settings and volumerollofftype, as they were unused, and changed the default mixer from muffled to master and the default rolloff to linear from logarithmic.
